index.html

```diff
@@ -1644,6 +1644,72 @@ <h2>
     </section>
   </section>
   </section>
+
+  <section>
+    <h2>Verification Algorithm</h2>
+    <p>
+This specification might be used with many different key discovery protocols.
+As such, discovery of verification keys is described in a different section of this document, 
+and is assumed to have succeeded prior to beginning the verification process.
```

Comment:
This is exactly what I'm hoping for. Do you think you can make it that verifiers MUST NOT attempt to discover or retrieve keys during the process of verifying signatures or issuers?

Reply:
We can state something like that... although I am not sure how normative we can make it....

All verification keys are assumed to have been previously dereferenced, keys SHOULD not be resolved at verification time, resolving verification keys at verification time leaks metadata.

The challenge is that for the 3 party model, verifiers do not know up front every issuer or holder they will encounter.